client/signup/steps/import-url/index.jsx

```diff
@@ -0,0 +1,124 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { localize } from 'i18n-calypso';
+import { connect } from 'react-redux';
+import { debounce, flow, get } from 'lodash';
+import debugFactory from 'debug';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import ButtonGroup from 'components/button-group';
+import StepWrapper from 'signup/step-wrapper';
+import SignupActions from 'lib/signup/actions';
+import FormTextInputWithAction from 'components/forms/form-text-input-with-action';
+//import FormInputValidation from 'components/forms/form-input-validation';
+import { fetchIsSiteImportable, setNuxUrlInputValue } from 'state/importer-nux/actions';
+import {
+	getNuxUrlInputValue,
+	getSiteDetails,
+	isUrlInputDisabled,
+} from 'state/importer-nux/temp-selectors';
+
+const debug = debugFactory( 'calypso:signup-step-import-url' );
+
+const DEBOUNCE_CHECK_INTERVAL = 1200;
+
+const normalizeUrlForImportSource = url => {
+	// @TODO sanitize? Prepend https:// ..?
+	return url;
+};
+
+class ImportURLStepComponent extends Component {
+	componentDidMount() {
+		const { queryObject } = this.props;
+		const urlFromQueryArg = normalizeUrlForImportSource( get( queryObject, 'url' ) );
+
+		if ( urlFromQueryArg ) {
+			this.handleInputChange( urlFromQueryArg );
+		}
+	}
+
+	handleAction = importUrl => {
+		event.preventDefault();
+		debug( { importUrl } );
+
+		SignupActions.submitSignupStep( { stepName: this.props.stepName }, [], {
+			importUrl,
+			themeSlugWithRepo: 'pub/radcliffe-2',
+		} );
+
+		this.props.goToNextStep();
+	};
+
+	handleInputChange = value => {
+		this.props.setNuxUrlInputValue( value );
+		this.debouncedFetchIsSiteImportable( normalizeUrlForImportSource( value ) );
+	};
+
+	fetchIsSiteImportable = () => {
+		const normalizedUrl = normalizeUrlForImportSource( this.props.urlInputValue );
+		this.props.fetchIsSiteImportable( normalizedUrl );
+	};
+
+	debouncedFetchIsSiteImportable = debounce( this.fetchIsSiteImportable, DEBOUNCE_CHECK_INTERVAL );
+
+	renderContent = () => {
+		const { isInputDisabled, urlInputValue, translate } = this.props;
+
+		return (
+			<div className="import-url__wrapper">
+				<FormTextInputWithAction
+					placeholder="Enter the URL of your existing site"
+					action="Continue"
+					onAction={ this.handleAction }
+					label={ translate( 'URL' ) }
+					onChange={ this.handleInputChange }
+					disabled={ isInputDisabled }
+					value={ urlInputValue }
+				/>
+				{ /* <FormInputValidation text="..." /> */ }
+				<ButtonGroup>
+					<Button>Skip</Button>
```

Comment:
Note: not wired up. I think this can also be [handled by the `StepWrapper` component](https://github.com/Automattic/wp-calypso/blob/59bdfeeb97eda4266ad39410cb0a074d2c88dbc8/client/signup/step-wrapper/index.jsx#L24).

Reply:
removed -- thx